### DISCUSSION TITLE
Request for support of lumi.switch.acn032

### DISCUSSION THREAD

**Discussion by sll1001** [merdok/homebridge-miot#640] [General]
I have 2 of these Aqara S1E smart switch installed.
After adding these to homebridge-miot config, they appear as intended, i.e. each device has 3 switches.
However, the naming of the switches are "Left Switch Service" "Middle Switch Service" and "Right Switch Service". Because there are 2 devices, there are duplicated "Left Switch Service", "Middle Switch Service" and "Right Switch Service". This cause strange behavior with the operation of the switches.

I tried to use-- "custom accessory": true -- to create a switch to dis-ambiguous the name, something like this:

     {
            "name": "Living SWS1",
            "ip": "172.16.88.83",
            "token": "xxxxx",
            "deviceId": "xxxxxxx",
            "pollingInterval": 10,
            "deepDebugLog": true,
            "silentLog": false,
            "deviceEnabled": true,
            "customAccessory": false,
            "onlyMainService": false,
            "buzzerControl": false,
            "ledControl": false,
            "childLockControl": false,
            "modeControl": false,
            "suppressAutoServiceCreation": [
                "temperature",
                "relativeHumidity",
                "illumination"
            ],
            "propertyControl": [
                {
                    "property": "4.1",
                    "name": "Living SWS1 R"
                }
            ]
        }, 

Indeed a switch "Living SWS1 R" is created. But miot keeps reporting polling Living SWS1 failed. Seems like some polling property is missing.

Could you suggest what I should add to make polling works using custom accessory?

Any other suggestions to get it to disambiguous the name of the switches?

**Comment by sll1001** [comment]
Some more details.

With just one smart switch device and without adding propertyControl, 3 switches are created.
Controlling the switches in IOS home app works, sort of. The switches state seem to change randomly.

**Reply by merdok** [reply]
I think that you need to create action buttons to control the respective switch toggle action. At least that is what I could see from the spec

**Reply by sll1001** [reply]
Thank you for your reply. I look deeper into the code and found 2 bugs (which I have filed) that causes the behaviour. This is a multi-switch with a touch LCD panel. I made some temporary fix to the code and manage to get it working properly. Please see how to write the proper code to fix the bug.

**Reply by merdok** [reply]
Thanks for opening the issue, I will try to provide a fix.
